### PR TITLE
testcase/filesystem: Add smartfs device mount path

### PR DESCRIPTION
**Pull request by lokeshbv**
Adds smartfs device mount path of lm3s6965-ek board in itc fs tc.

Signed-off-by: Lokesh B V <lokesh.bv@partner.samsung.com>

### CODE REVIEW

**Comment by sunghan-chang** [comment]
@lokeshbv Please fix coding rule violation first.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c [hunk 1]
```diff
@@ -66,9 +66,10 @@
 #ifdef CONFIG_FS_SMARTFS
 #if defined(CONFIG_SIDK_S5JT200_AUTOMOUNT_USERFS)
 #define TMP_MOUNT_DEV_DIR CONFIG_SIDK_S5JT200_AUTOMOUNT_USERFS_DEVNAME
-
 #elif defined(CONFIG_ARTIK05X_AUTOMOUNT_USERFS)
 #define TMP_MOUNT_DEV_DIR CONFIG_ARTIK05X_AUTOMOUNT_USERFS_DEVNAME
+#elif defined(CONFIG_ARCH_BOARD_LM3S6965EK)
+#define TMP_MOUNT_DEV_DIR "/dev/smart0p0"
```
Comment (sunghan-chang) [inline apps/examples/testcase/le_tc/filesystem/fs_main.c]: Is it fixed always as smart0p0?

**Comment by sunghan-chang** [comment]
@lokeshbv Please ignore my previous comment. You didn't make it. We will do it later.

File: apps/examples/testcase/le_tc/filesystem/fs_main.c (hunk 1 above)
Comment (lokeshbv) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: It is created using minor no. and part no. and they are not configurable. So I hard coded it here.
@manoharahk kindly share your opinion about this change

Comment (pradeepns85) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: I believe, it's better to provide this interface through defconfig as it remains a common place to refer for used device name and minor no for all kinds of file systems

Comment (sunghan-chang) [reply apps/examples/testcase/le_tc/filesystem/fs_main.c]: @lokeshbv @pradeepns85 I agree Pradeep's suggestion. We had better one file for fs mount point interface. Now, it is defined at each location with different types.

**Comment by seinfra** [comment]
Target : [f4fbe283619114f3c7b7747d769f606092d20d40] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [f4fbe283619114f3c7b7747d769f606092d20d40] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9c0a81b5a6c2f2e410b185e432c786b5b2168a3d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [43afbb70581a01e100d2debbaabd9e11707af9dc] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [43afbb70581a01e100d2debbaabd9e11707af9dc] - Code Rule Check OK.

File: os/arch/arm/src/lm3s6965-ek/Kconfig [hunk 2]
```diff
@@ -24,4 +24,12 @@ config QEMU_NONE
 	---help---
 		Dont build for Qemu Hardware.
 endchoice
+
+config LM3S_AUTOMOUNT_USERFS_DEVNAME
```
Comment (sunghan-chang) [inline os/arch/arm/src/lm3s6965-ek/Kconfig]: There is no code to support this feature.
Is it ok to test fs?

**Comment by seinfra** [comment]
Target : [a6fccca5f9967615f0bfb0e742a750ada3dd8d96] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a6fccca5f9967615f0bfb0e742a750ada3dd8d96] - Code Rule Check OK.

**Pull request merged**